Add docs (#428)
rancher/rke2: docs/policies.md

* If ran with the `cis-1.5` profile, RKE2 will apply a restrictive policy called `global-restricted-psp` to all namespaces except `kube-system`. The `kube-system` namespace needs a less restrictive policy named `system-unrestricted-psp` in order to launch critical components.
* If ran without the `cis-1.5` profile, RKE2 will apply a completely unrestricted policy called `global-unrestricted-psp`, which is the equivalent of running without the PSP admission controller enabled.

RKE2 will put these policies in place upon initial startup, but will not modify them after that, unless explicitly triggered by the cluster operator as described below. This is to allow the operator to fully control the PSPs without RKE2's defaults adding interference.

Creation and application of the PSPs is controlled by the presence or absence of certain annotations on the `kube-system` namespace. These map directly to the PSPs which can be created and are:

 * `psp.rke2.io/global-restricted`
 * `psp.rke2.io/system-unrestricted`
 * `psp.rke2.io/global-unrestricted`

The following logic is performed at startup for the policies and their annotations:

> * If the annotation exists, RKE2 continues without further action

Reviewer: ```suggestion
* If the annotation exists, RKE2 continues without further action.
```

> * If the annotations doesn't exist, RKE2 checks to see if the associated policy exists and if so, deletes and recreates it, along with adding the annotation to the namespace.

Reviewer: ```suggestion
* If the annotation doesn't exist, RKE2 checks to see if the associated policy exists and if so, deletes and recreates it, along with adding the annotation to the namespace.
```

* In the case of the `global-unrestricted-psp`, the policy is not recreated. This is to account for moving between CIS and non-CIS modes without making the cluster less secure.
* At the time of creating a policy, cluster roles and cluster role bindings are also created to ensure the appropriate policies are put into use by default.

> So, after initial startup, operators can modify or delete RKE2's policies and RKE2 will respect those changes. Additionally, to "reset" a policy, an operator just needs to delete the associated annotation from the `kube-system` namespace and restart RKE2.

Reviewer: Start-up is hyphonated previously. Not sure if you want it here too for consistency.

```suggestion
So, after initial start-up, operators can modify or delete RKE2's policies and RKE2 will respect those changes. Additionally, to "reset" a policy, an operator just needs to delete the associated annotation from the `kube-system` namespace and restart RKE2.
```

  - Ingress
```

> RKE2 applies 2 network policy for all built-in namespaces and updates the namespaces with annotations `np.rke2.io` and `np.rke2.io/dns`.

Reviewer: All built-in namespaces get the one policy but only kube-system gets the dns policy.

Author: So should this read
```
RKE2 applies the `default-network-policy` policy and `np.rke2.io` annotation to all built-in namespaces. The `kube-system` namespace additionally gets the `default-network-dns-policy` policy and `np.rke2.io/dns` annotation applied to it.
```
?

Reviewer: Yeah, that works.
